pkg/validator/bool_pointer_helpers.go

```diff
@@ -0,0 +1,55 @@
+package validator
+
```

Comment:
Instead of all this, I think we can get away with a single function:

```go
func boolValue(boolPtr *bool, default bool) bool {
  if boolPtr == nil {
    return default
  }
  return *boolPtr
}
```

(note: you can do `nil` comparison without casting to a bool - I know we were casting before but I don't think that buys us anything.)

Then you can do e.g. `getBoolVal(securityContext.RunAsNonRoot, getBoolVal(podSecurityContext.RunAsNonRoot, false))`

Can probably just put it in `container.go` if it works.